cmd/influxd/launcher/query_test.go

```diff
@@ -745,3 +748,79 @@ from(bucket: "%s")
 		t.Fatal(err)
 	}
 }
+
+func TestLauncher_Query_PushDownWindowAggregateCount(t *testing.T) {
+	l := launcher.RunTestLauncherOrFail(t, ctx, "--feature-flags", "pushDownWindowAggregateCount=true")
+	l.SetupOrFail(t)
+	defer l.ShutdownOrFail(t, ctx)
+
+	l.WritePointsOrFail(t, `
+m0,k=k0 f=0i 0
+m0,k=k0 f=1i 1000000000
+m0,k=k0 f=2i 2000000000
+m0,k=k0 f=3i 3000000000
+m0,k=k0 f=4i 4000000000
+m0,k=k0 f=5i 5000000000
+m0,k=k0 f=6i 6000000000
+m0,k=k0 f=5i 7000000000
+m0,k=k0 f=0i 8000000000
+m0,k=k0 f=6i 9000000000
+m0,k=k0 f=6i 10000000000
+m0,k=k0 f=7i 11000000000
+m0,k=k0 f=5i 12000000000
+m0,k=k0 f=8i 13000000000
+m0,k=k0 f=9i 14000000000
+m0,k=k0 f=5i 15000000000
+`)
+
+	query := fmt.Sprintf(`
+from(bucket: "%s")
+	|> range(start: 1970-01-01T00:00:00Z, stop: 1970-01-01T00:00:15Z)
+	|> aggregateWindow(every: 5s, fn: count)
+	|> drop(columns: ["_start", "_stop"])
+`, l.Bucket.Name)
+	res := l.MustExecuteQuery(query)
+	defer res.Done()
+	got := flux.NewSliceResultIterator(res.Results)
+	defer got.Release()
+
+	dec := csv.NewMultiResultDecoder(csv.ResultDecoderConfig{})
+	data := `
+#datatype,string,long,long,string,string,string,dateTime:RFC3339
+#group,false,false,false,true,true,true,false
+#default,_result,,,,,,
+,result,table,_value,_field,_measurement,k,_time
+,,0,5,f,m0,k0,1970-01-01T00:00:05Z
+,,0,5,f,m0,k0,1970-01-01T00:00:10Z
+,,0,5,f,m0,k0,1970-01-01T00:00:15Z
+`
+	want, err := dec.Decode(ioutil.NopCloser(strings.NewReader(data)))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer want.Release()
+
+	if err := executetest.EqualResultIterators(want, got); err != nil {
+		t.Fatal(err)
+	}
+
+	h, ok := func() (*dto.Histogram, bool) {
+		const metricName = "query_influxdb_source_read_request_duration_seconds"
+		mf := l.Metrics(t)[metricName]
+		for _, m := range mf.Metric {
+			for _, label := range m.Label {
+				if label.GetName() == "op" && label.GetValue() == "readWindowAggregate" {
+					return m.GetHistogram(), true
```

Comment:
Did you write this as a function just to avoid using a label?

Reply:
I think I did it just to avoid the `var found bool` pattern. I don't think I realized that this was a nested for loop at the time. I just like to scope my local variables as much as possible when I can.